parsers/pascal.c

```diff
@@ -38,10 +38,16 @@ static kindDefinition PascalKinds [] = {
 */
 
 static void createPascalTag (
-		tagEntryInfo* const tag, const vString* const name, const int kind)
+		tagEntryInfo* const tag, const vString* const name, const int kind,
+		const char *arglist, const char *vartype)
 {
 	if (PascalKinds [kind].enabled  &&  name != NULL  &&  vStringLength (name) > 0)
+	{
 		initTagEntry (tag, vStringValue (name), kind);
+		tag->extensionFields.signature = arglist;
+		tag->extensionFields.typeRef[0] = eStrdup ("typename");
```

Comment:
Is it correct to dup the string here?
I found varying uses in other parsers and so I am unsure whether it should be done or not. It seems to work without as well but I might have just been lucky.

Same for the assignment of `typeRef[1]` below.

Reply:
In this case, you should not do "dup".
"dup" is needed once a tag entry is copied to the cork queue.
See https://docs.ctags.io/en/latest/internal.html?highlight=cork#output-tag-stream-1.

If the cork queue is enabled in a parser, a tag passed makeTagEntry is copied to a memory buffer named "cork queue". makeTagEntry returns an `id` on the queue. You can get back the tag entry with passing the `id` to getEntryInCorkQueue(id). After getting back, you can update a member of the tag entry. When updating the member, you must do "dup".

The pascal parser doesn't use the cork queue. So you don't have to do "dup".

If you can use valgrind, you can test the mem leak with "make units LANGUAGES=Pascal VG=1".

Reply:
Thanks for detailed explanation.
I checked with Valgrind locally in the meantime and noticed there also that the string duplication was not necessary.
Squashed and pushed the fix.